index.js

```diff
@@ -70,23 +71,43 @@ class GithubScm extends Scm {
     /**
     * Constructor
     * @method constructor
-    * @param  {Object} options           Configuration options
-    * @param  {Object} options.retry     Configuration options for circuit breaker retries
-    * @param  {Object} options.breaker   Configuration options for circuit breaker
+    * @param  {Object} options                      Configuration options
+    * @param  {String}  [options.gheHost=null]      If using GitHub Enterprise, the host/port of the deployed instance
+    * @param  {String}  [options.gheProtocol=https] If using GitHub Enterprise, the protocol to use
+    * @param  {Boolean} [options.https=false]       Is the Screwdriver API running over HTTPS
+    * @param  {String}  options.oauthClientId       OAuth Client ID provided by GitHub application
+    * @param  {String}  options.oauthClientSecret   OAuth Client Secret provided by GitHub application
+    * @param  {Object}  [options.fusebox={}]        Circuit Breaker configuration
     * @return {GithubScm}
     */
-    constructor(config) {
+    constructor(config = {}) {
         super();
 
-        this.config = config;
-        this.github = new Github();
+        // Validate configuration
+        this.config = joi.attempt(config, joi.object().keys({
+            gheProtocol: joi.string().optional().default('https'),
+            gheHost: joi.string().optional().description('GitHub Enterpise host'),
+            https: joi.boolean().optional().default(false),
+            oauthClientId: joi.string().required(),
+            oauthClientSecret: joi.string().required(),
+            fusebox: joi.object().default({})
+        }).unknown(true), 'Invalid config for GitHub');
+
+        const githubConfig = {};
+
+        if (this.config.gheHost) {
+            githubConfig.host = this.config.gheHost;
+            githubConfig.protocol = this.config.gheProtocol;
+            githubConfig.pathPrefix = '/api/v3';
```

Comment:
For GHE, it's the prefix for all API paths. [Source](https://developer.github.com/v3/enterprise/#endpoint-urls)